termine-be/cli.py

```diff
@@ -296,3 +297,124 @@ def load_frontend_config(db: directives.PeeweeSession, frontend_config_file: hug
 
                 config.save()
                 print("Done.")
+
+@hug.cli(output=hug.output_format.pretty_json)
+def get_bookings_created_at(db: directives.PeeweeSession, booked_at: hug.types.text):
+    """
+    Get bookings for a day with yyyy-mm-dd or one specific booking at yyyy-mm-ddThh:mm:ss.mmmmmm
+    """
+    with db.atomic():
+        query = Booking.select(
+            Booking,
+            Appointment.time_slot.start_date_time.alias("start_date_time")
+        ).join(Appointment).join(TimeSlot)
+
+        booked_start = datetime.fromisoformat(booked_at).replace(tzinfo=None)
+
+        if str(booked_start.date()) == booked_at:
+            # booked_at is yyyy-mm-dd
+            booked_end = booked_start.date() + timedelta(days=1)
+            bookings = query.where(Booking.booked_at.between(booked_start, booked_end))
+        else:
+            # booked_at is yyyy-mm-ddThh:mm:ss.mmmmmm
+            bookings = query.where(Booking.booked_at == booked_start)
+
+        result = []
+        for booking in bookings.dicts().iterator():
+            del booking["appointment"]
+            result.append({**booking})
+
+        return result
+
+
+@hug.cli(output=hug.output_format.pretty_json)
+def free_slots_at(db: directives.PeeweeSession, user: hug.types.text, at_datetime: hug.types.text = None):
+    """
+    args: USER_NAME [--at_datetime ISO_DATETIME]
+    """
+    free_slots = api.next_free_slots(db, User.get(User.user_name == user), at_datetime)
+    start_date = datetime.fromisoformat(at_datetime).replace(tzinfo=None)
+
+    slots = []
+    for slot in free_slots["slots"]:
+        if slot["startDateTime"].date() == start_date.date():
+            slot["startDateTime"] = str(slot["startDateTime"])
+            slots.append(slot)
+
+    return slots
+
+
+@hug.cli(output=hug.output_format.pretty_json)
+def claim_appointment(db: directives.PeeweeSession, start_date_time: hug.types.text, user: hug.types.text):
+    """
+    args: (ISO_DATE | ISO_DATETIME) USER_NAME
+    """
+    try:
+        api_claim_appointment = api.claim_appointment(
+            db, start_date_time, User.get(User.user_name == user)
+        )
+    except hug.HTTPGone as e:
+        return None
+
+    return api_claim_appointment
+
+
+@hug.cli(output=hug.output_format.pretty_json)
+def has_booked_appointment(db: directives.PeeweeSession, user: hug.types.text):
+    """
+    args: USER_NAME
+    """
+    return Booking.select(Booking).where(Booking.booked_by == user).count() > 0
+
+
+@hug.cli(output=hug.output_format.pretty_json)
+def book_followup(db: directives.PeeweeSession, booking: hug.types.json):
+    """
+    args: BOOKING_JSON
+    """
+    if has_booked_appointment(db, booking["booked_by"]):
```

Comment:
This is probably wrong. At least after this command was ran once, there exists a booking made by that user and all following bookings will be discarded.